apps/pipelines/nodes/base.py

```diff
@@ -45,6 +45,22 @@ def add_temp_state_messages(left: dict, right: dict):
     return output
 
 
+def merge_dicts(left: dict, right: dict):
+    """
+    Merge two dictionaries, combining values for the same key into a list. The value of any key is expected to be a list
+    """
+    output = {**left}
+    for key, value in right.items():
+        if key in output:
+            if isinstance(output[key], list):
+                output[key] = list(set(output[key]) | set(value))
```

Comment:
what made you decide to merge using sets?

Reply:
To deduplicate the values